bigtable/snippets/src/test/java/com/example/bigtable/FiltersTest.java

```diff
@@ -0,0 +1,738 @@
+                    + "Reading data for phone#5c10102#20190501\n"
+                    + "Column Family cell_plan\n"
+                    + "\tdata_plan_10gb: true @%1$s\n"
+                    + "Column Family stats_summary\n"
+                    + "\tconnected_cell: \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0001 @%1$s\n\n"
+                    + "Reading data for phone#5c10102#20190502\n"
+                    + "Column Family cell_plan\n"
+                    + "\tdata_plan_10gb: true @%1$s\n"
+                    + "Column Family stats_summary\n"
+                    + "\tconnected_cell: \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0001 @%1$s\n",
+                TIMESTAMP, TIMESTAMP - HOUR));
+  }
+
+  @Test
+  public void testFilterLimitCellsPerRowOffset() {
+    Filters.filterLimitCellsPerRowOffset(projectId, instanceId, TABLE_ID);
+
+    String output = bout.toString();
+    assertThat(output)
+        .contains(
+            String.format(
+                "Reading data for phone#4c410523#20190501\n"
+                    + "Column Family cell_plan\n"
+                    + "\tdata_plan_05gb: true @%1$s\n"
+                    + "Column Family stats_summary\n"
+                    + "\tconnected_cell: \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0001 @%1$s\n"
+                    + "\tconnected_wifi: \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0001 @%1$s\n"
+                    + "\tos_build: PQ2A.190405.003 @%1$s\n\n"
+                    + "Reading data for phone#4c410523#20190502\n"
+                    + "Column Family stats_summary\n"
+                    + "\tconnected_wifi: \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0001 @%1$s\n"
+                    + "\tos_build: PQ2A.190405.004 @%1$s\n\n"
+                    + "Reading data for phone#4c410523#20190505\n"
+                    + "Column Family stats_summary\n"
+                    + "\tconnected_wifi: \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0001 @%1$s\n"
+                    + "\tos_build: PQ2A.190406.000 @%1$s\n\n"
+                    + "Reading data for phone#5c10102#20190501\n"
+                    + "Column Family stats_summary\n"
+                    + "\tconnected_wifi: \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0001 @%1$s\n"
+                    + "\tos_build: PQ2A.190401.002 @%1$s\n\n"
+                    + "Reading data for phone#5c10102#20190502\n"
+                    + "Column Family stats_summary\n"
+                    + "\tconnected_wifi: \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 @%1$s\n"
+                    + "\tos_build: PQ2A.190406.000 @%1$s",
+                TIMESTAMP));
+  }
+
+  @Test
+  public void testFilterColFamilyRegex() {
+    Filters.filterLimitColFamilyRegex(projectId, instanceId, TABLE_ID);
+
+    String output = bout.toString();
+    assertThat(output)
+        .contains(
+            String.format(
+                "Reading data for phone#4c410523#20190501\n"
```

Comment:
Is it possible to define a regex for a single entry and verify it applies X number of times instead?

Reply:
That wouldn't really test this accurately then since there are various combinations which would give the same number of cells that are different